crates/primitives-traits/src/header/sealed.rs

```diff
@@ -65,6 +65,11 @@ impl<H: alloy_consensus::BlockHeader> SealedHeader<H> {
     pub fn num_hash(&self) -> BlockNumHash {
         BlockNumHash::new(self.number(), self.hash)
     }
+
+    /// Return a [`BlockWithParent`] for this header.
+    pub fn block_with_parent(&self) -> BlockWithParent {
+        BlockWithParent { parent: self.parent_hash(), block: self.num_hash() }
```

Comment:
it's only available for the concrete `Header` as `num_hash_with_parent_slow`

We can't enforce this in `BlockHeader` because we can't require `Sealable` as it's undefined for `AnyNetwork`'s header

Reply:
afaict it does not exist in `alloy_consensus::BlockHeader`, but would be a good thing to add

Reply:
> We can't enforce this in BlockHeader because we can't require Sealable as it's undefined for AnyNetwork's header

Ah I see

Reply:
ah right -.-

Reply:
`hash_with_parent_slow` wouldn't really be useful here I believe? we already have the hash on `SealedHeader`, so we'd anyway need some kind of helper directly on it to avoid hashing

we could introduce an extension trait for `Sealed<H: BlockHeader>` in alloy but it's not useful for now until we have a standalone type for it